### ISSUE TITLE
[entgql] gql_where_input P() got a rewrite cycle detected

### ISSUE THREAD

**Issue by tsingsun** [ent/ent#2472]
i have a project with many shema and got `gql_where_input` 5M size, when I run debug in vscode .got the compile error: 

```txt
Starting: [/Users/tsingsun/go/bin/dlv]() dap --check-go-version=false --listen=127.0.0.1:57334 --log-dest=3 from [/Users/tsingsun/workspace/project/datacenter/indicator/cmd/web]()
DAP server listening at: 127.0.0.1:57334
Build Error: go build -o [/Users/tsingsun/workspace/project/datacenter/indicator/cmd/web/__debug_bin]() -gcflags all=-N -l .
# t.qeelyn.com[/datacenter/indicator/ent]()
.[./../ent/gql_where_input.go:2019:14](): internal compiler error: '(*FinBalanceSheetWhereInput).P': rewrite cycle detected

Please file a bug report including a short program that triggers the error.
[https://go.dev/issue/new]() (exit status 2)
```

 2019 line see blow

```
// P returns a predicate for filtering finbalancesheets.
// An error is returned if the input is empty or invalid.
func (i *FinBalanceSheetWhereInput) P() (predicate.FinBalanceSheet, error) {
	var predicates []predicate.FinBalanceSheet
	if i.Not != nil {  // **line: 2019**
		p, err := i.Not.P()
		if err != nil {
			return nil, err
		}
		predicates = append(predicates, finbalancesheet.Not(p))
	}
	switch n := len(i.Or); {
	case n == 1:
		p, err := i.Or[0].P()
		if err != nil {
			return nil, err
		}
		predicates = append(predicates, p)
	case n > 1:
		or := make([]predicate.FinBalanceSheet, 0, n)
		for _, w := range i.Or {
			p, err := w.P()
			if err != nil {
				return nil, err
			}
			or = append(or, p)
		}
		predicates = append(predicates, finbalancesheet.Or(or...))
	}
.... line 2017
.......
....line 6948
}
```
the FinBalanceSheetWhereInput.P() almost have 3000  line code.

- [x] The issue is present in the latest release.
- [x] I have searched the [issues](https://github.com/ent/ent/issues) of this repository and believe that this is not a duplicate.

## Current Behavior 😯

1. Use Run Build `go build -o ./cmd/web/serve-darwin ./cmd/web/main.go` the programe RUN good.

2. Use Debug Build `go build -o  -gcflags all=-N -l ` got compile error:

3. the other project with less shemas and smaller gql_where_input .  RUN and DEBUG  are good

4. go 1.17 all is ok5. 

## Expected Behavior 🤔

the generate code debug ok, although maybe it's  a compiler-level problem,

## Steps to Reproduce 🕹

Steps:

1. make gql_where_input large
2.
3.
4.

## Your Environment 🌎

| Tech        | Version |
| ----------- | ------- |
| Go          | 1.18.1  |
| Ent         | 0.10.1 laste commit   |
| Database    | MySQL   |
| Driver      | https://github.com/go-sql-driver/mysql        |

**Comment by randall77** [comment]
Could you give us instructions on how to download this project, and exactly what build command you used?
I got as far as `git clone https://github.com/ent/contrib` but I don't know what to do after that to reproduce the problem.
(e.g. there's no `.../web/main.go` file in that repository.)

**Comment by giautm** [comment]
Hello @tsingsun, can you create a reproduce repository with https://github.com/ent/bug, please?

**Comment by tsingsun** [comment]
The project is private, so I think about how to build a new project to reproduce the problem.

**Comment by tsingsun** [comment]
@giautm @randall77 reproduce repository : https://github.com/tsingsun/ent-issue-2472.git

in vscode locate `server/server.go` and `F5` run will see the problem.

**Comment by randall77** [comment]
Could you figure out what command-line commands reproduce the problem? I'm not a vscode user and the error I get when I hit F5 on the server.go page isn't the one you're reporting.

> build entgo.io[/bug/server](): cannot load embed: malformed module path "embed": missing dot in first path element

**Comment by tsingsun** [comment]
you run the go build in the project root directory
```
go build -o ./server/__debug_bin -gcflags all=-"N -l" ./server 
```

**Comment by randall77** [comment]
Ok, I can reproduce. I think this is just #51639. For some reason that issue didn't get requested for a backport to 1.18, I'll do that now.

**Comment by tsingsun** [comment]
> you run the go build in the project root directory
> 
> ```
> go build -o ./server/__debug_bin -gcflags all=-"N -l" ./server 
> ```

seems resolved in go 1.18.2

**Comment by a8m** [comment]
Great. Thanks for the update @tsingsun 

Can we close this?
